api/v1/services/TokenService.js

```diff
@@ -0,0 +1,45 @@
+var Token = require('../models/Token');
+var config = require('../../config');
+var logger = require('../../logging');
+
+/**
+ * Finds a token given the Token value
+ * @param {String} value The Token's value
+ * @param {String} scope The Scope the token is for
+ * @return {Promise} resolving to the associated Token Model
+ * @throws {NotFoundError} when the requested token cannot be found
+ */
+module.exports.findTokenByValue = function(value, scope) {
+	return Token
+		.findByValue(value)
+		.then(function(result) {
+			if (!result) {
+				var message = "Could not find the provided token to reset password";
+				var source = "token";
+				throw new errors.InvalidParameterError(message, source);
+			}
+
+			// Provided logging for the DEFAULT case in case the developer did not intend to use it
+			var defaultExpiration = function() {
```

Comment:
You should be throwing a `TypeError`, not passing it through. If you want to keep the default scope, that's fine, maybe it can be useful. But we should be treating omissions as errors.